js/api/posts/reactions.mjs

```
@@ -0,0 +1,22 @@
+import { POSTS_URL, headers } from "../auth/constants.mjs";
+
+// Send reaction to the API 
+export async function addReaction(post) {
+    try {
+        const response = await fetch(`${POSTS_URL}/${post.postId}/react/👍`, {
+            method: "PUT",
+            body: JSON.stringify(post),
+            headers: headers(),
+        });
+        
+        if (!response.ok) {
+            throw new Error("Failed to add reaction");
+        }
+        
+        const responseData = await response.json();
+        return responseData;
+    } catch (error) {
+        console.error("Error adding reaction:", error);
+        throw error;
```

`catch block`: to suppress an error
`throw`: to create an error

Understand the differences, they have different purposes. Here you're basically catching an error and throwing it again.

**Suggestion:** remove the throw from the catch block and use the try/catch block in a UI/UX code block e.g event listener.